homeassistant/components/reolink/binary_sensor.py

```diff
@@ -0,0 +1,174 @@
+    ),
+    ReolinkBinarySensorDescription(
+        key=VEHICLE_DETECTION_TYPE,
+        name=VEHICLE_DETECTION_TYPE,
+        icon="mdi:car",
+        icon_off="mdi:car-off",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.ai_detected(ch, VEHICLE_DETECTION_TYPE),
+        supported=lambda host, ch: host.api.ai_supported(ch, VEHICLE_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorDescription(
+        key=PET_DETECTION_TYPE,
+        name=PET_DETECTION_TYPE,
+        icon="mdi:dog-side",
+        icon_off="mdi:dog-side-off",
+        device_class=BinarySensorDeviceClass.MOTION,
+        value=lambda host, ch: host.api.ai_detected(ch, PET_DETECTION_TYPE),
+        supported=lambda host, ch: host.api.ai_supported(ch, PET_DETECTION_TYPE),
+    ),
+    ReolinkBinarySensorDescription(
+        key="visitor",
+        name="Visitor",
+        icon="mdi:bell-ring-outline",
+        icon_off="mdi:doorbell",
+        value=lambda host, ch: host.api.visitor_detected(ch),
+        supported=lambda host, ch: host.api.is_doorbell_enabled(ch),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Reolink IP Camera."""
+    reolink_data: ReolinkData = hass.data[DOMAIN][config_entry.entry_id]
+    host = reolink_data.host
+
+    entities = []
+    for channel in host.api.channels:
+        for entity_description in BINARY_SENSORS:
+            if not entity_description.supported(host, channel):
+                continue
+
+            entities.append(
+                ReolinkBinarySensorEntity(
+                    reolink_data,
+                    channel,
+                    entity_description,
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class ReolinkBinarySensorEntity(ReolinkCoordinatorEntity, BinarySensorEntity):
+    """An implementation of a base binary-sensor class for Reolink IP camera motion sensors."""
+
+    _attr_has_entity_name = True
+    entity_description: ReolinkBinarySensorDescription
+
+    def __init__(
+        self,
+        reolink_data: ReolinkData,
+        channel: int,
+        entity_description: ReolinkBinarySensorDescription,
+    ) -> None:
+        """Initialize Reolink binary sensor."""
+        ReolinkCoordinatorEntity.__init__(self, reolink_data, channel)
+        self.entity_description = entity_description
+
+        self._attr_name = entity_description.name
+        self._attr_unique_id = (
+            f"{self._host.unique_id}_{self._channel}_{entity_description.key}"
+        )
+        self._attr_device_class = entity_description.device_class
+
+    @property
+    def icon(self) -> str:
+        """Icon of the sensor."""
+        if self.entity_description.value(self._host, self._channel):
+            return self.entity_description.icon
+
+        return self.entity_description.icon_off
```

Comment:
Handle the `off` case, let the default icon handling of HA do it otherwise.

```suggestion
        if not self.entity_description.value(self._host, self._channel):
            return self.entity_description.icon_off
		return super().icon
```